core/connection/manager.go

```diff
@@ -82,10 +82,12 @@ type connectionManager struct {
 
 	//these are populated by Connect at runtime
 	ctx             context.Context
-	mutex           sync.RWMutex
 	status          Status
 	sessionInfo     SessionInfo
 	cleanConnection func()
+
+	statusLock sync.RWMutex
```

Comment:
should go right after field it is guarding. That why sometimes field name can be not used in its mutex.